__tests__/router.spec.ts

```diff
@@ -70,6 +69,13 @@ describe('Router', () => {
     )
   })
 
+  it('can allows the end use to override parseQuery', async () => {
+    const customQueryParser = jest.fn()
+    const { router } = await newRouter({ parseQuery: customQueryParser })
+    router.resolve('/foo?bar=baz')
+    expect(customQueryParser).toHaveBeenCalled()
```

Comment:
Let's call it `parseQuery` so it can just be passed around like this and also let's check it's called with the right parameters
```suggestion
  it('can override parseQuery', async () => {
    const parseQuery = jest.fn()
    const { router } = await newRouter({ parseQuery })
    router.resolve('/foo?bar=baz')
    expect(parseQuery).toHaveBeenCalledWith('?bar=baz')
```